### ISSUE TITLE
tarfile.TarFile.errors is initialized for some reason as None even though it really only accepts str

### ISSUE THREAD

**Issue by kbaikov** [python/cpython#127424]
# Bug report

### Bug description:

See the exact line here: https://github.com/python/cpython/blob/ee57221872ad009df648e22d2f113f907607ec16/Lib/tarfile.py#L1687

After that the errors is used only as string in encode/decode. And is initialized in the dunder init with default `errors="surrogateescape"`.

So perhaps it would be a good idea to use `errors=""` in #L1687 ? or directly with the default `errors="surrogateescape"` ?

This was discovered as part of the typeshed, where `tarfile.TarFile.errors` was added into the allowed list since mypy complained: https://github.com/python/typeshed/blob/2ccc53bb6716463680fcf1d204642c46299b7f88/stdlib/%40tests/stubtest_allowlists/common.txt#L46C1-L46C16

I will add a PR here and to the typeshed if this gets approved.

### CPython versions tested on:

3.13

### Operating systems tested on:

Windows

**Comment by sobolevn** [comment]
I don't think that this is really a bug. It more looks like a refactoring. Can it affect existing users?

**Comment by picnixz** [comment]
> I think it is initialized to None and there seems to be nothing wrong with it, after all, int and str are a class.

This is not the issue. The issue concerns typing and the fact that `errors` must be typed `str | None` in this case, instead of just `str`. As for whether having a `None` value makes sense at runtime except for default-detection, the question remains and may affect existing users as Nikita hinted.
